WalletWasabi.Tests/UnitTests/Bases/LastExceptionTrackerTests.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using WalletWasabi.Bases;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Bases
+{
+	public class LastExceptionTrackerTests
+	{
+		[Fact]
+		public void ProcessTest()
+		{
+			var let = new LastExceptionTracker();
+
+			// No exception was processed at this point.
+			Assert.Null(let.LastException);
```

Comment:
This is leaking abstraction. LastException shouldn't be visible from outside.

Reply:
In my design `LastExceptionTracker` is a standalone class and it can be used whenever user wants to track exceptions. Also it helps me to test the code.